meta/store.go

```diff
@@ -151,9 +171,44 @@ func (s *Store) IDPath() string { return filepath.Join(s.path, "id") }
 
 // Open opens and initializes the raft store.
 func (s *Store) Open() error {
+	// If we have a join addr, attempt to join
+	if s.join != "" {
+
+		joined := false
+		for _, join := range strings.Split(s.join, ",") {
+			res, err := s.rpc.join(s.Addr.String(), join)
+			if err != nil {
+				s.Logger.Printf("join failed: %v", err)
+				continue
+			}
+			joined = true
```

Comment:
It seems the only point of this variable is for the log message and return at the end. Yet there are other ways this function can return, in error, within this function.

How about:

```
if i == len(strings.Split(s.join, ",")) {
    return fmt.Errorf("failed to join existing cluster at %v", s.join)
}
``` 
within the `err != nil` block?